opl/consumer_lag.py

```diff
@@ -12,26 +12,50 @@ class ConsumerLag:
     bootstrap_server and kafka group as input.
     """
 
-    def __init__(self, topic, bootstrap_servers, group) -> None:
+    def __init__(
+        self, topic, bootstrap_servers, group, username="", password=""
+    ) -> None:
         self.topic = topic
         self.group = group
         self.bootstrap_servers = bootstrap_servers
         self.logger = logging.getLogger("consumer_lag")
         self.offset_records = {}
+        self.username = username
+        self.password = password
 
     def _getconsumer(self):
-        consumer = KafkaConsumer(
-            bootstrap_servers=self.bootstrap_servers,
-            auto_offset_reset="latest",
-            enable_auto_commit=False,
-            max_poll_records=50,
-            max_poll_interval_ms=300000,
-            group_id=self.group,
-            session_timeout_ms=50000,
-            heartbeat_interval_ms=10000,
-            consumer_timeout_ms=100000,
-        )
+        # Common parameters for both cases
+        common_params = {
+            "bootstrap_servers": self.bootstrap_servers,
+            "auto_offset_reset": "latest",
+            "enable_auto_commit": False,
+            "max_poll_records": 50,
+            "max_poll_interval_ms": 300000,
+            "group_id": self.group,
+            "session_timeout_ms": 50000,
+            "heartbeat_interval_ms": 10000,
+            "consumer_timeout_ms": 100000,
+        }
 
+        # Kafka consumer creation: SASL or noauth
+        try:
```

Comment:
With this try we will print first log info message every time. Could you replace the `try:` directly with `if self.username != "":`?

Reply:
Or maybe `if self.username is not None:` given how default is specified in `add_kafka_opts(`)

Reply:
Or maybe I should set the add_kafka_opts default to ""?